### ISSUE TITLE
FR: [Ingestion] - Synchronize parts information with OpenSearch index (6 of 6)

### ISSUE THREAD

**Issue by urtzi-sotes** [aemsites/vg-roadchoice-com#45]
### Context/Description

The OpenSearch index must be synchronized with the parts' information whenever a part is created, updated, or deleted. Thus, we need to implement a mechanism for synchronization.
A repository_dispatch event is sent to the GitHub repo whenever a file is published or unpublished. We can create a GitHub action to trigger the ingestion pipeline whenever the main parts' file is published:

- [/product-data/road-choice-make-model-part-filter-options.json](https://www.roadchoice.com/product-data/road-choice-make-model-part-filter-options.json)

We must ensure that all events are processed sequentially.

### Acceptance Criteria

- The mechanism to perform synchronization is defined and documented in the wiki.
- The mechanism is implemented.
- The ingestion pipeline is triggered every time one of the related files is published as preview.
- end to end testing:
  - Every time a part file is published, the pipeline is triggered via de GitHub action.
  - As a result, the OpenSearch index is updated and:
    - new products are searchable.
    - deleted products are no longer searchable.
    - updated products contain the updated values

### Out of scope

- Queue service to process all events sequentially

**Comment by ZoeBeykirch** [comment]
All ingestion tickets need to be completed before this one can be picked up.

**Comment by ZoeBeykirch** [comment]
Once this is completed, then can deploy the ingestion pipeline

**Comment by urtzi-sotes** [comment]
The synchronization mechanism has been implemented and deployed to production.

end to end testing has been done in production by publishing the `/product-data/road-choice-make-model-part-filter-options.json` file and the workflow works as expected:
  - the pipeline is triggered via de GitHub action.
  - the OpenSearch index is updated and:
       - new products are searchable.
       - deleted products are no longer searchable.
       - updated products contain the updated values
